### PR TITLE
feat(spanner): enable request options for batch read

### LINKED ISSUE #3904: spanner: Batch read transactions miss support for RPC Priority and Tagging
Batch read transactions miss `WithOptions` methods that allow an application to pass in an `RpcPriority` and/or tags for the requests.

First reported in #3896

### PR DESCRIPTION
**Pull request by olavloite**
Enables RequestOptions (priority and tagging) for batch read transactions. This problem was first reported in #3896.

cc @yfuruyama

Fixes #3904

### CODE REVIEW

**Review by hengfengli — COMMENTED** [review]
LGTM. @skuruppu But do we need to hold off this change because it is related to tagging?

**Comment by skuruppu** [comment]
I think because the PR is already merged for tagging, we should merge this in for completeness.

**Review by hengfengli — APPROVED** [review]

**Comment by olavloite** [comment]
@skuruppu Apparently this PR can only be merged once a code owner has approved.

**Pull request merged**
